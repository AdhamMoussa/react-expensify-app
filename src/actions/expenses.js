import database from "../firebase/firebase";
import moment from "moment";

export const addExpense = expense => {
  return {
    type: "ADD_EXPENSE",
    newExpense: {
      ...expense
    }
  };
};

export const startAddExpense = (expenseData = {}) => {
  return dispatch => {
    const {
      description = "",
      amount = 0,
      note = "",
      createdAt = 0
    } = expenseData;
    const expense = { description, amount, note, createdAt };
    return database
      .ref("expenses")
      .push(expense)
      .then(ref => {
        dispatch(
          addExpense({
            id: ref.key,
            ...expense
          })
        );
      });
  };
};

export const editExpense = (id, updatedData) => {
  return {
    type: "EDIT_EXPENSE",
    id,
    updatedData
  };
};

export const startEditExpense = (id, updatedData) => {
  return dispatch => {
    return database
      .ref(`expenses/${id}`)
      .update({ ...updatedData })
      .then(() => {
        dispatch(
          editExpense(id, {
            ...updatedData,
            createdAt: moment(updatedData.createdAt)
          })
        );
      });
  };
};

export const removeExpense = id => {
  return {
    type: "REMOVE_EXPENSE",
    id
  };
};

export const startRemoveExpense = id => {
  return dispatch => {
    return database
      .ref(`expenses/${id}`)
      .remove()
      .then(() => {
        dispatch(removeExpense(id));
      });
  };
};

export const setExpenses = expenses => {
  return {
    type: "SET_EXPENSES",
    expenses
  };
};

export const startSetExpenses = () => {
  return dispatch => {
    return database
      .ref("expenses")
      .once("value")
      .then(snapshot => {
        const expenses = [];
        snapshot.forEach(child => {
          expenses.push({
            id: child.key,
            ...child.val(),
            createdAt: moment(child.val().createdAt)
          });
        });
        dispatch(setExpenses(expenses));
      });
  };
};
