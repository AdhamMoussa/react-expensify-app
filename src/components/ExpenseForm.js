import React from "react";
import moment from "moment";
import "react-dates/initialize";
import { SingleDatePicker } from "react-dates";

class ExpenseForm extends React.Component {
  state = {
    description: this.props.editedExpense
      ? this.props.editedExpense.description
      : "",
    amount: this.props.editedExpense ? this.props.editedExpense.amount : "",
    note: this.props.editedExpense ? this.props.editedExpense.note : "",
    createdAt: this.props.editedExpense
      ? this.props.editedExpense.createdAt
      : moment(),
    datePickerFocused: false,
    descriptionError: false,
    amountError: false
  };
  handleDescriptionChange = e => {
    const description = e.target.value;
    this.setState(() => ({ description }));
  };
  handleAmountChange = e => {
    const amount = e.target.value;
    if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/))
      this.setState(() => ({ amount }));
  };
  handleNoteChange = e => {
    const note = e.target.value;
    this.setState(() => ({ note }));
  };
  handleFormSubmit = e => {
    e.preventDefault();
    // handle description error
    if (!this.state.description) {
      this.setState(() => ({
        descriptionError: true
      }));
    } else if (this.state.description && this.state.descriptionError) {
      this.setState(() => ({ descriptionError: false }));
    }
    // handle amount error
    if (!this.state.amount) {
      this.setState(() => ({
        amountError: true
      }));
    } else if (this.state.amount && this.state.amountError) {
      this.setState(() => ({ amountError: false }));
    }
    // Submit
    if (this.state.description && this.state.amount) {
      this.setState(() => ({
        descriptionError: false,
        amountError: false
      }));
      this.props.onSubmitHandler({
        description: this.state.description,
        amount: parseFloat(this.state.amount),
        createdAt: this.state.createdAt.valueOf(),
        note: this.state.note
      });
    }
  };
  render() {
    return (
      <div>
        <form className="form" onSubmit={this.handleFormSubmit}>
          <div className="form__group">
            <input
              id="description-input"
              type="text"
              name="description"
              value={this.state.description}
              onChange={this.handleDescriptionChange}
              placeholder="Expense Description"
              className="form__input"
              autoFocus
            />
            <p
              className={
                this.state.descriptionError
                  ? "error-msg error-msg--visible"
                  : "error-msg"
              }
            >
              Please insert a valid description of your expense
            </p>
          </div>
          <div className="form__group">
            <input
              id="amount-input"
              type="text"
              name="amount"
              value={this.state.amount}
              onChange={this.handleAmountChange}
              placeholder="Amount"
              className="form__input"
            />
            <p
              className={
                this.state.amountError
                  ? "error-msg error-msg--visible"
                  : "error-msg"
              }
            >
              Please insert a valid amount of your expense
            </p>
          </div>
          <div className="form__group">
            <SingleDatePicker
              date={this.state.createdAt}
              onDateChange={date => {
                if (date) {
                  this.setState({ createdAt: date });
                }
              }}
              focused={this.state.datePickerFocused}
              onFocusChange={({ focused }) =>
                this.setState({ datePickerFocused: focused })
              }
              numberOfMonths={1}
              isOutsideRange={day => false}
            />
          </div>
          <div className="form__group">
            <textarea
              id="note-input"
              name="note"
              value={this.state.note}
              onChange={this.handleNoteChange}
              placeholder="Expense Note"
              className="form__input"
            />
          </div>
          <div className="form__group">
            <button className="button" type="submit">
              Submit
            </button>
          </div>
        </form>
      </div>
    );
  }
}

export default ExpenseForm;
