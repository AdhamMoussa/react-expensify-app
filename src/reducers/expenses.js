const defaultExpensesState = [];

const expensesReducer = (state = defaultExpensesState, action) => {
  switch (action.type) {
    case "ADD_EXPENSE":
      return [...state, action.newExpense];
    case "EDIT_EXPENSE":
      return state.map(
        expense =>
          expense.id === action.id
            ? { ...expense, ...action.updatedData }
            : expense
      );
    case "REMOVE_EXPENSE":
      return state.filter(expense => expense.id !== action.id);
    default:
      return state;
  }
};

export default expensesReducer;
