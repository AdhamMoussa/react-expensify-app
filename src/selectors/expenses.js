import moment from 'moment';

const getExpensesView = (expenses, filters) => {
  return expenses.filter((expense) => {
    const textMatch = expense.description.toLowerCase().includes(filters.text.toLowerCase());
    const startDateMatch = filters.startDate ? expense.createdAt.isSameOrAfter(filters.startDate) : true;
    const endDateMatch = filters.endDate ? expense.createdAt.isSameOrBefore(filters.endDate) : true;

    return textMatch && startDateMatch && endDateMatch;
  }).sort((a, b) => {
    if (filters.sortBy === 'date') {
      return a.createdAt.isAfter(b.createdAt) ? -1 : 1;
    } else if (filters.sortBy === 'amount') {
      return a.amount > b.amount ? -1 : 1;
    }
  });
};

export default getExpensesView;
