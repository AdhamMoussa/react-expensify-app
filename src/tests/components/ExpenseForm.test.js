import React from 'react';
import { shallow } from 'enzyme';
import getType from 'jest-get-type';
import ExpenseForm from '../../components/ExpenseForm';
import { SingleDatePicker } from 'react-dates';
import expenses from '../fixtures/demoExpenses';
import moment from 'moment';

test('should render ExpenseForm correctly', () => {
  const wrapper = shallow(<ExpenseForm />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseForm with data correctly', () => {
  const wrapper = shallow(<ExpenseForm editedExpense={expenses[1]} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render error for invalid description & amount inputs', () => {
  const wrapper = shallow(<ExpenseForm />);
  expect(wrapper).toMatchSnapshot();
  wrapper.find('form').simulate('submit', {
    preventDefault: () => {}
  });
  expect(getType(wrapper.state('descriptionError'))).toBe('string');
  expect(getType(wrapper.state('amountError'))).toBe('string');
  expect(wrapper).toMatchSnapshot();
});

test('should set description on input change', () => {
  const value = 'some text';
  const wrapper = shallow(<ExpenseForm />);
  wrapper.find('#description-input').simulate('change', {
    target: { value }
  });
  expect(wrapper.state('description')).toBe(value);
});

test('should set amount on input change', () => {
  const value = '452.15';
  const wrapper = shallow(<ExpenseForm />);
  wrapper.find('#amount-input').simulate('change', {
    target: { value }
  });
  expect(wrapper.state('amount')).toBe(value);
});

test('should NOT set amount state when input is invalid', () => {
  const value = '452.222';
  const wrapper = shallow(<ExpenseForm />);
  wrapper.find('#amount-input').simulate('change', {
    target: { value }
  });
  expect(wrapper.state('amount')).toBe('');
});

test('should set note state on input change', () => {
  const value = 'note text';
  const wrapper = shallow(<ExpenseForm />);
  wrapper.find('#note-input').simulate('change', {
    target: { value }
  });
  expect(wrapper.state('note')).toBe(value);
});

test('should call onSubmit props for valid form submission', () => {
  const onSubmitSpy = jest.fn();
  const wrapper = shallow(<ExpenseForm editedExpense={expenses[1]} onSubmitHandler={onSubmitSpy} />);
  wrapper.find('form').simulate('submit', {
    preventDefault: () => {}
  });
  delete expenses[1].id;
  expect(onSubmitSpy).toHaveBeenLastCalledWith(expenses[1]);
});

test('should set createdAt state on date change', () => {
  const now = moment();
  const wrapper = shallow(<ExpenseForm />);
  wrapper.find(SingleDatePicker).prop('onDateChange')(now);
  expect(wrapper.state('createdAt')).toEqual(now);
});

test('should set datePickerFocused state on focus change', () => {
  const focused = true;
  const wrapper = shallow(<ExpenseForm />);
  wrapper.find(SingleDatePicker).prop('onFocusChange')({focused});
  expect(wrapper.state('datePickerFocused')).toBe(focused);
});












