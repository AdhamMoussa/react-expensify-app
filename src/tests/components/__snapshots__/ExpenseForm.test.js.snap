// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseForm correctly 1`] = `
<div>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <div
      className="form__group"
    >
      <input
        autoFocus={true}
        className="form__input"
        id="description-input"
        name="description"
        onChange={[Function]}
        placeholder="Expense Description"
        type="text"
        value=""
      />
      <p
        className="error-msg"
      >
        Please insert a valid description of your expense
      </p>
    </div>
    <div
      className="form__group"
    >
      <input
        className="form__input"
        id="amount-input"
        name="amount"
        onChange={[Function]}
        placeholder="Amount"
        type="text"
        value=""
      />
      <p
        className="error-msg"
      >
        Please insert a valid amount of your expense
      </p>
    </div>
    <div
      className="form__group"
    >
      <withStyles(SingleDatePicker)
        anchorDirection="left"
        appendToBody={false}
        block={false}
        calendarInfoPosition="bottom"
        customCloseIcon={null}
        customInputIcon={null}
        date={"1970-01-01T00:00:00.000Z"}
        daySize={39}
        disableScroll={false}
        disabled={false}
        displayFormat={[Function]}
        enableOutsideDays={false}
        firstDayOfWeek={null}
        focused={false}
        hideKeyboardShortcutsPanel={false}
        horizontalMargin={0}
        horizontalMonthPadding={13}
        id="date"
        initialVisibleMonth={null}
        inputIconPosition="before"
        isDayBlocked={[Function]}
        isDayHighlighted={[Function]}
        isOutsideRange={[Function]}
        isRTL={false}
        keepFocusOnInput={false}
        keepOpenOnDateSelect={false}
        monthFormat="MMMM YYYY"
        navNext={null}
        navPrev={null}
        noBorder={false}
        numberOfMonths={1}
        onClose={[Function]}
        onDateChange={[Function]}
        onFocusChange={[Function]}
        onNextMonthClick={[Function]}
        onPrevMonthClick={[Function]}
        openDirection="down"
        orientation="horizontal"
        phrases={
          Object {
            "calendarLabel": "Calendar",
            "chooseAvailableDate": [Function],
            "clearDate": "Clear Date",
            "closeDatePicker": "Close",
            "dateIsSelected": [Function],
            "dateIsUnavailable": [Function],
            "enterKey": "Enter key",
            "escape": "Escape key",
            "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
            "homeEnd": "Home and end keys",
            "jumpToNextMonth": "Move forward to switch to the next month.",
            "jumpToPrevMonth": "Move backward to switch to the previous month.",
            "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
            "keyboardShortcuts": "Keyboard Shortcuts",
            "leftArrowRightArrow": "Right and left arrow keys",
            "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
            "moveFocusByOneMonth": "Switch months.",
            "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
            "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
            "openThisPanel": "Open this panel.",
            "pageUpPageDown": "page up and page down keys",
            "questionMark": "Question mark",
            "returnFocusToInput": "Return to the date input field.",
            "selectFocusedDate": "Select the date in focus.",
            "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
            "upArrowDownArrow": "up and down arrow keys",
          }
        }
        placeholder="Date"
        readOnly={false}
        regular={false}
        renderCalendarInfo={null}
        renderDayContents={null}
        renderMonthElement={null}
        renderMonthText={null}
        reopenPickerOnClearDate={false}
        required={false}
        screenReaderInputMessage=""
        showClearDate={false}
        showDefaultInputIcon={false}
        small={false}
        verticalHeight={null}
        verticalSpacing={22}
        weekDayFormat="dd"
        withFullScreenPortal={false}
        withPortal={false}
      />
    </div>
    <div
      className="form__group"
    >
      <textarea
        className="form__input"
        id="note-input"
        name="note"
        onChange={[Function]}
        placeholder="Expense Note"
        value=""
      />
    </div>
    <div
      className="form__group"
    >
      <button
        className="button"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;

exports[`should render ExpenseForm with data correctly 1`] = `
<div>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <div
      className="form__group"
    >
      <input
        autoFocus={true}
        className="form__input"
        id="description-input"
        name="description"
        onChange={[Function]}
        placeholder="Expense Description"
        type="text"
        value="water"
      />
      <p
        className="error-msg"
      >
        Please insert a valid description of your expense
      </p>
    </div>
    <div
      className="form__group"
    >
      <input
        className="form__input"
        id="amount-input"
        name="amount"
        onChange={[Function]}
        placeholder="Amount"
        type="text"
        value={190}
      />
      <p
        className="error-msg"
      >
        Please insert a valid amount of your expense
      </p>
    </div>
    <div
      className="form__group"
    >
      <withStyles(SingleDatePicker)
        anchorDirection="left"
        appendToBody={false}
        block={false}
        calendarInfoPosition="bottom"
        customCloseIcon={null}
        customInputIcon={null}
        date={"1969-12-31T23:59:59.000Z"}
        daySize={39}
        disableScroll={false}
        disabled={false}
        displayFormat={[Function]}
        enableOutsideDays={false}
        firstDayOfWeek={null}
        focused={false}
        hideKeyboardShortcutsPanel={false}
        horizontalMargin={0}
        horizontalMonthPadding={13}
        id="date"
        initialVisibleMonth={null}
        inputIconPosition="before"
        isDayBlocked={[Function]}
        isDayHighlighted={[Function]}
        isOutsideRange={[Function]}
        isRTL={false}
        keepFocusOnInput={false}
        keepOpenOnDateSelect={false}
        monthFormat="MMMM YYYY"
        navNext={null}
        navPrev={null}
        noBorder={false}
        numberOfMonths={1}
        onClose={[Function]}
        onDateChange={[Function]}
        onFocusChange={[Function]}
        onNextMonthClick={[Function]}
        onPrevMonthClick={[Function]}
        openDirection="down"
        orientation="horizontal"
        phrases={
          Object {
            "calendarLabel": "Calendar",
            "chooseAvailableDate": [Function],
            "clearDate": "Clear Date",
            "closeDatePicker": "Close",
            "dateIsSelected": [Function],
            "dateIsUnavailable": [Function],
            "enterKey": "Enter key",
            "escape": "Escape key",
            "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
            "homeEnd": "Home and end keys",
            "jumpToNextMonth": "Move forward to switch to the next month.",
            "jumpToPrevMonth": "Move backward to switch to the previous month.",
            "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
            "keyboardShortcuts": "Keyboard Shortcuts",
            "leftArrowRightArrow": "Right and left arrow keys",
            "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
            "moveFocusByOneMonth": "Switch months.",
            "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
            "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
            "openThisPanel": "Open this panel.",
            "pageUpPageDown": "page up and page down keys",
            "questionMark": "Question mark",
            "returnFocusToInput": "Return to the date input field.",
            "selectFocusedDate": "Select the date in focus.",
            "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
            "upArrowDownArrow": "up and down arrow keys",
          }
        }
        placeholder="Date"
        readOnly={false}
        regular={false}
        renderCalendarInfo={null}
        renderDayContents={null}
        renderMonthElement={null}
        renderMonthText={null}
        reopenPickerOnClearDate={false}
        required={false}
        screenReaderInputMessage=""
        showClearDate={false}
        showDefaultInputIcon={false}
        small={false}
        verticalHeight={null}
        verticalSpacing={22}
        weekDayFormat="dd"
        withFullScreenPortal={false}
        withPortal={false}
      />
    </div>
    <div
      className="form__group"
    >
      <textarea
        className="form__input"
        id="note-input"
        name="note"
        onChange={[Function]}
        placeholder="Expense Note"
        value=""
      />
    </div>
    <div
      className="form__group"
    >
      <button
        className="button"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;

exports[`should render error for invalid description & amount inputs 1`] = `
<div>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <div
      className="form__group"
    >
      <input
        autoFocus={true}
        className="form__input"
        id="description-input"
        name="description"
        onChange={[Function]}
        placeholder="Expense Description"
        type="text"
        value=""
      />
      <p
        className="error-msg"
      >
        Please insert a valid description of your expense
      </p>
    </div>
    <div
      className="form__group"
    >
      <input
        className="form__input"
        id="amount-input"
        name="amount"
        onChange={[Function]}
        placeholder="Amount"
        type="text"
        value=""
      />
      <p
        className="error-msg"
      >
        Please insert a valid amount of your expense
      </p>
    </div>
    <div
      className="form__group"
    >
      <withStyles(SingleDatePicker)
        anchorDirection="left"
        appendToBody={false}
        block={false}
        calendarInfoPosition="bottom"
        customCloseIcon={null}
        customInputIcon={null}
        date={"1970-01-01T00:00:00.000Z"}
        daySize={39}
        disableScroll={false}
        disabled={false}
        displayFormat={[Function]}
        enableOutsideDays={false}
        firstDayOfWeek={null}
        focused={false}
        hideKeyboardShortcutsPanel={false}
        horizontalMargin={0}
        horizontalMonthPadding={13}
        id="date"
        initialVisibleMonth={null}
        inputIconPosition="before"
        isDayBlocked={[Function]}
        isDayHighlighted={[Function]}
        isOutsideRange={[Function]}
        isRTL={false}
        keepFocusOnInput={false}
        keepOpenOnDateSelect={false}
        monthFormat="MMMM YYYY"
        navNext={null}
        navPrev={null}
        noBorder={false}
        numberOfMonths={1}
        onClose={[Function]}
        onDateChange={[Function]}
        onFocusChange={[Function]}
        onNextMonthClick={[Function]}
        onPrevMonthClick={[Function]}
        openDirection="down"
        orientation="horizontal"
        phrases={
          Object {
            "calendarLabel": "Calendar",
            "chooseAvailableDate": [Function],
            "clearDate": "Clear Date",
            "closeDatePicker": "Close",
            "dateIsSelected": [Function],
            "dateIsUnavailable": [Function],
            "enterKey": "Enter key",
            "escape": "Escape key",
            "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
            "homeEnd": "Home and end keys",
            "jumpToNextMonth": "Move forward to switch to the next month.",
            "jumpToPrevMonth": "Move backward to switch to the previous month.",
            "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
            "keyboardShortcuts": "Keyboard Shortcuts",
            "leftArrowRightArrow": "Right and left arrow keys",
            "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
            "moveFocusByOneMonth": "Switch months.",
            "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
            "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
            "openThisPanel": "Open this panel.",
            "pageUpPageDown": "page up and page down keys",
            "questionMark": "Question mark",
            "returnFocusToInput": "Return to the date input field.",
            "selectFocusedDate": "Select the date in focus.",
            "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
            "upArrowDownArrow": "up and down arrow keys",
          }
        }
        placeholder="Date"
        readOnly={false}
        regular={false}
        renderCalendarInfo={null}
        renderDayContents={null}
        renderMonthElement={null}
        renderMonthText={null}
        reopenPickerOnClearDate={false}
        required={false}
        screenReaderInputMessage=""
        showClearDate={false}
        showDefaultInputIcon={false}
        small={false}
        verticalHeight={null}
        verticalSpacing={22}
        weekDayFormat="dd"
        withFullScreenPortal={false}
        withPortal={false}
      />
    </div>
    <div
      className="form__group"
    >
      <textarea
        className="form__input"
        id="note-input"
        name="note"
        onChange={[Function]}
        placeholder="Expense Note"
        value=""
      />
    </div>
    <div
      className="form__group"
    >
      <button
        className="button"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;

exports[`should render error for invalid description & amount inputs 2`] = `
<div>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <div
      className="form__group"
    >
      <input
        autoFocus={true}
        className="form__input"
        id="description-input"
        name="description"
        onChange={[Function]}
        placeholder="Expense Description"
        type="text"
        value=""
      />
      <p
        className="error-msg error-msg--visible"
      >
        Please insert a valid description of your expense
      </p>
    </div>
    <div
      className="form__group"
    >
      <input
        className="form__input"
        id="amount-input"
        name="amount"
        onChange={[Function]}
        placeholder="Amount"
        type="text"
        value=""
      />
      <p
        className="error-msg error-msg--visible"
      >
        Please insert a valid amount of your expense
      </p>
    </div>
    <div
      className="form__group"
    >
      <withStyles(SingleDatePicker)
        anchorDirection="left"
        appendToBody={false}
        block={false}
        calendarInfoPosition="bottom"
        customCloseIcon={null}
        customInputIcon={null}
        date={"1970-01-01T00:00:00.000Z"}
        daySize={39}
        disableScroll={false}
        disabled={false}
        displayFormat={[Function]}
        enableOutsideDays={false}
        firstDayOfWeek={null}
        focused={false}
        hideKeyboardShortcutsPanel={false}
        horizontalMargin={0}
        horizontalMonthPadding={13}
        id="date"
        initialVisibleMonth={null}
        inputIconPosition="before"
        isDayBlocked={[Function]}
        isDayHighlighted={[Function]}
        isOutsideRange={[Function]}
        isRTL={false}
        keepFocusOnInput={false}
        keepOpenOnDateSelect={false}
        monthFormat="MMMM YYYY"
        navNext={null}
        navPrev={null}
        noBorder={false}
        numberOfMonths={1}
        onClose={[Function]}
        onDateChange={[Function]}
        onFocusChange={[Function]}
        onNextMonthClick={[Function]}
        onPrevMonthClick={[Function]}
        openDirection="down"
        orientation="horizontal"
        phrases={
          Object {
            "calendarLabel": "Calendar",
            "chooseAvailableDate": [Function],
            "clearDate": "Clear Date",
            "closeDatePicker": "Close",
            "dateIsSelected": [Function],
            "dateIsUnavailable": [Function],
            "enterKey": "Enter key",
            "escape": "Escape key",
            "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
            "homeEnd": "Home and end keys",
            "jumpToNextMonth": "Move forward to switch to the next month.",
            "jumpToPrevMonth": "Move backward to switch to the previous month.",
            "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
            "keyboardShortcuts": "Keyboard Shortcuts",
            "leftArrowRightArrow": "Right and left arrow keys",
            "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
            "moveFocusByOneMonth": "Switch months.",
            "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
            "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
            "openThisPanel": "Open this panel.",
            "pageUpPageDown": "page up and page down keys",
            "questionMark": "Question mark",
            "returnFocusToInput": "Return to the date input field.",
            "selectFocusedDate": "Select the date in focus.",
            "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
            "upArrowDownArrow": "up and down arrow keys",
          }
        }
        placeholder="Date"
        readOnly={false}
        regular={false}
        renderCalendarInfo={null}
        renderDayContents={null}
        renderMonthElement={null}
        renderMonthText={null}
        reopenPickerOnClearDate={false}
        required={false}
        screenReaderInputMessage=""
        showClearDate={false}
        showDefaultInputIcon={false}
        small={false}
        verticalHeight={null}
        verticalSpacing={22}
        weekDayFormat="dd"
        withFullScreenPortal={false}
        withPortal={false}
      />
    </div>
    <div
      className="form__group"
    >
      <textarea
        className="form__input"
        id="note-input"
        name="note"
        onChange={[Function]}
        placeholder="Expense Note"
        value=""
      />
    </div>
    <div
      className="form__group"
    >
      <button
        className="button"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;
